### PR TITLE
Boost 1 67

### PR DESCRIPTION
**Pull request by artemp**

### CODE REVIEW

File: include/mapnik/expression_grammar_x3_def.hpp [hunk 1]
```diff
@@ -70,6 +70,16 @@ namespace mapnik { namespace grammar {
     auto const& escaped_unicode = json::grammar::escaped_unicode;
     }
 
+    template <typename Context>
+    inline mapnik::transcoder const& extract_transcoder(Context const& ctx)
+    {
+#if BOOST_VERSION >= 106700
+        return x3::get<transcoder_tag>(ctx);
+#else
+        return x3::get<transcoder_tag>(ctx).get();
```
Comment (lightmare) [inline include/mapnik/expression_grammar_x3_def.hpp]: this branch is unnecessary. `reference_wrapper` is implicitly convertible to the stored reference

File: include/mapnik/expression_grammar_x3_def.hpp [hunk 2]
```diff
@@ -130,7 +140,7 @@ namespace mapnik { namespace grammar {
 
     auto do_unicode = [] (auto const& ctx)
     {
-        auto const& tr = x3::get<transcoder_tag>(ctx).get();
+        auto const& tr = extract_transcoder(ctx);
```
Comment (lightmare) [inline include/mapnik/expression_grammar_x3_def.hpp]:
```c++
mapnik::transcoder const& tr = x3::get<transcoder_tag>(ctx);
```

File: include/mapnik/expression_grammar_x3_def.hpp (hunk 1 above)
Comment (artemp) [reply include/mapnik/expression_grammar_x3_def.hpp]: :+1:

File: include/mapnik/expression_grammar_x3_def.hpp (hunk 2 above)
Comment (artemp) [reply include/mapnik/expression_grammar_x3_def.hpp]: yes, fully spelling out returned type works ^ but I prefer helper function still, specially when it used multiple time.

**Comment by lightmare** [comment]
I wonder, wouldn't it be easier to stick with `reference_wrapper` and avoid the conditionals?

**Pull request merged**